lib/asyn-thread.c

```diff
@@ -170,7 +170,7 @@ void destroy_thread_sync_data(struct thread_sync_data *tsd)
    * close one end of the socket pair (may be done in resolver thread);
    * the other end (for reading) is always closed in the parent thread.
    */
-#ifndef HAVE_EVENTFD
+#if !defined(HAVE_EVENTFD) || !defined(HAVE_SYS_EVENTFD_H)
```

Comment:
How about we do:

~~~c
#if defined(HAVE_EVENTFD) && defined(HAVE_SYS_EVENTFD_H)
#define USE_EVENTFD
#endif
~~~

in a central location instead and then use that single define in check?